doc/guidelines/e2e_frontend_coding_guidelines.md

```diff
@@ -0,0 +1,145 @@
+
+Nevertheless the exception might occur where getByRole() and getByTestId() don't fit the case. In that scenario a proper decision between the team must be made.
+
+#### [getByRole](https://playwright.dev/docs/locators#locate-by-role)
+
+> To make tests resilient, we recommend prioritizing user-facing attributes and explicit contracts such as page.getByRole()
+
+Allows locating elements by their ARIA role, ARIA attributes and accessible name.
+
+#### [getByTestId](https://playwright.dev/docs/locators#locate-by-test-id) (data-testid)
+
+> Testing by test ids is the most resilient way of testing as even if your text or role of the attribute changes the test will still pass. QA's and developers should define explicit test ids and query them with [page.getByTestId()](https://playwright.dev/docs/api/class-page#page-get-by-test-id). However testing by test ids is not user facing. If the role or text value is important to you then consider using user facing locators such as [role](https://playwright.dev/docs/locators#locate-by-role) and [text locators](https://playwright.dev/docs/locators#locate-by-text).
+
+Locate element by `data-testid`.
+The most resilient attribute.
+
+#### [getByLabel](https://playwright.dev/docs/locators#locate-by-label)
+
+> Most form controls usually have dedicated labels that could be conveniently used to interact with the form. In this case, you can locate the control by its associated label using [page.getByLabel()](https://playwright.dev/docs/api/class-page#page-get-by-label).
+
+Allows locating input elements by the text of the associated <label> or aria-labelledby element, or by the aria-label attribute.
+This is a reasonable alternative popular for input forms.
+
+#### [getByText](https://playwright.dev/docs/locators#locate-by-text)
+
+> We recommend using text locators to find non interactive elements like div, span, p, etc. For interactive elements like button, a, input, etc. use [role locators](https://playwright.dev/docs/locators#locate-by-role).
+
+Allows locating elements that contain given text.
+This alternative should be picked only if the other three are not possible.
+It is not very resilient and can frequently break the tests.
+
+#### [Playwright suggested locators](https://playwright.dev/docs/input)
+
+- text inputs: **label**
+- checkboxes: **label**
+- radio buttons: **label**
+- select options: **label**
+- mouse clicks: **role or text**
+- type characters: **locator**
+- keys and shortcuts: **text or role** **<- this is the only place where text is fine**
+- upload files and focus elements: **label**
+
+### 2. User facing or resilience?
+
+When chosing the right locator it is important to ask yourself one question.
+Do we want to test what the user perceives?
+If so, go with `roles`.
+If user-centricity is not so important on this element use `data-testid`.
+In case of doubt use `data-testid`.
+
+### 3. Naming Conventions
```

Comment:
As written in #639 I think we should have one document for both `data-testid` and what we use for internationalization, so this should be in a separate document so that we can refer it from both places.